staging/src/k8s.io/client-go/tools/cache/reflector.go

```diff
@@ -188,7 +185,10 @@ func (r *Reflector) ListAndWatch(stopCh <-chan struct{}) error {
 	klog.V(3).Infof("Listing and watching %v from %s", r.expectedTypeName, r.name)
 	var resourceVersion string
 
-	options := metav1.ListOptions{ResourceVersion: r.relistResourceVersion()}
+	// Explicitly set "0" as resource version - it's fine for the List()
```

Comment:
Just to avoid loosing my comment in the PR body, let me put it here (with Joe`s response):

>> @jpbetz @liggitt - do we want to revert it on master too? I thought that we will just revert in 1.17 branch.

> Good question. I put this on master since that's the process I've always followed for fixes that I cherrypick, but in this case we're going to want this enabled on master so we can follow up with the fix forward of #86430 so maybe it's okay to just revert on 1.17? What do you think @liggitt ?

Reply:
do we have tests demonstrating the issue this revert is intended to fix? where are those tests run (master? release branch?) and how are they run (automatically? manually?)

Since we intend to fix master for 1.18, the only reason I would want this in master is if it was needed to be picked up by tests. If we have sufficient test coverage on a 1.17-only pick, that seems ok.

Reply:
We don't have any automatic tests for it yet - https://github.com/kubernetes/kubernetes/issues/86484

So we run tests manually and we can run it against any branch/build.

Reply:
Given that, I'm ok just reverting in 1.17 as long as we run the tests against the release-1.17 branch.

Reply:
Sounds good. Open the PR against 1.17 now.